inspirehep/modules/records/serializers/fields_export.py

```diff
@@ -0,0 +1,318 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2014-2017 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""Instructions on where to find relevant bits of data in API."""
+
+from __future__ import absolute_import, division, print_function
+from idutils import is_arxiv_post_2007
+from inspire_utils.record import get_value
+from inspire_utils.date import PartialDate
+from inspirehep.modules.hal.utils import get_page_artid, get_conference_record, get_conference_title
+from babel import Locale
+from isbn_hyphenate import hyphenate, IsbnError
```

Comment:
Wow, somebody created an entire package just for this function?

Well, we already have an implementation in IDUtils of this: https://github.com/inveniosoftware/idutils/blob/f159a4105e3b26558c2e8fbae17818e3a1e2e81d/idutils/__init__.py#L447-L450, and we should probably get it from there. Or is it not considering some edge case?

Reply:
It's doing the opposite, the function you pointed to strips the formatting. Micha asked me to format the ISBNs into the hyphenated format, and it turned out that it is actually a complicated set of rules to do that, hence the whole package.